sdk/src/main/java/io/opentelemetry/sdk/trace/SpanWrapper.java

```diff
@@ -59,7 +59,7 @@ static SpanWrapper create(
       Status status) {
     return new AutoValue_SpanWrapper(
         delegate,
-        Collections.unmodifiableList(links),
+        links, // note: the caller has already made this unmodifiable
         Collections.unmodifiableList(events),
         Collections.unmodifiableMap(attributes),
```

Comment:
Let's be consistent and do the wrapping in one place.

Reply:
How do feel about using guava's ImmutableList here?

Reply:
Hm. It's also true that, currently, if the span hasn't ended, the attributes will still be modified by the running span, making the unmodifiable wrapper not really immutable. The only way I know to solve this is to copy the Map's contents, rather than just wrapping.

Reply:
That was the bug that I mentioned in the previous PR. You need to have a

```java
private HashMap<> getImmutableAttributes() {
  if (hasEnded) {
    return Collections.unmodifiableMap(attributes)
  }
  return Collections.unmodifiableMap(new HashMap<>(attributes))
}

```

Reply:
ok. I added a test for this particular case, and made it pass.

Reply:
(and moved the unmodifiable wrapping to all be in the RERS)